relay/pkg/block/forwarder.go

```diff
@@ -0,0 +1,138 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/tbtc/relay/pkg/btc"
+	"github.com/keep-network/tbtc/relay/pkg/chain"
+)
+
+const (
+	// Size of the headers queue.
+	headersQueueSize = 50
+
+	// Maximum size of processed headers batch.
+	headersBatchSize = 5
+
+	// Maximum time for which the pulling process will wait for a single header
+	// to be delivered by the headers queue.
+	headerTimeout = 1 * time.Second
+
+	// Block duration of a Bitcoin difficulty epoch.
+	difficultyEpochDuration = 2016
+
+	// Duration for which the forwarder should rest after performing
+	// a push action.
+	forwarderPushingSleepTime = 45 * time.Second
+)
+
+var logger = log.Logger("relay-block-forwarder")
```

Comment:
I'd go with `tbtc-relay` given that the repository is `tbtc`. We have `keep-beacon` logger for `keep-core/pkg/beacon` and `keep-tss` logger for `keep-ecdsa/pkg/ecdsa/tss`.

Reply:
I think it will be too long. We will have `tbtc-relay-block-forwarder`. The `tbtc` prefix doesn't give any value in my opinion. No need to know we are in the `tbtc` repo from the `relay`'s perspective.

Reply:
Per another comment, I'd really love to rename it to "relay" and then it's just `tbtc-relay`.

Reply:
So, maybe just `relay` then. Still not getting the point of forcing `tbtc` prefix here 😁

Reply:
To keep it consistent with keep client logging where all logs are prefixed with `keep-*` and it may help with filtering logs.
I was even thinking about `keep-relay` but the relay is a part of tBTC, not Keep Network.

Reply:
`tbtc` repo is part of `keep-network` organization so `keep-relay` is not that bad. But ok, `tbtc-relay` may also be the option.

Reply:
Resolving, this will be done in a separate PR according to the TODO: bce322f8b5881c410789c9c5f7a885815dbcc5f2